### PR TITLE
Fix the crash due to null image bindings in vkCopyDescriptorSet

### PR DESCRIPTION
**Pull request by yalcinmelihyasin**

### CODE REVIEW

**Comment by yalcinmelihyasin** [comment]
b/143274544

File: gapis/api/vulkan/api/descriptor.api [hunk 1]
```diff
@@ -661,26 +661,23 @@ cmd void vkUpdateDescriptorSets(
...
-        srcBinding.ImageBinding[c.SrcArrayIndex]
-        dstBinding.ImageBinding = imageBinding
 
         if srcBinding.ImageBinding[c.SrcArrayIndex] == null {
-          vkErrorInvalidSampler(null)
+           vkErrInvalidDescriptorArrayElement(as!u64(c.SrcSet), c.SrcBinding, c.SrcArrayIndex)
         } else {
+          imageBinding := dstBinding.ImageBinding
+          imageBinding[c.DstArrayIndex] =
+          srcBinding.ImageBinding[c.SrcArrayIndex]
```
Comment (pmuetschard) [inline gapis/api/vulkan/api/descriptor.api]: can you combine these two lines? i know they were separate before, but they really shouldn't be.... or at the very least indent the second one.

**Comment by chrisforbes** [comment]
:+1: confirmed this fixes the issue and we can now trace that workload

File: gapis/api/vulkan/api/descriptor.api (hunk 1 above)
Comment (yalcinmelihyasin) [reply gapis/api/vulkan/api/descriptor.api]: Surely.

**Comment by AWoloszyn** [comment]
In chatting offline, we want to make sure this isn't just hiding a different issue related to state reconstruction.

**Comment by yalcinmelihyasin** [comment]
I am merging this pr while still looking at the suspected underlying issue.

**Pull request merged**